packages/jest-util/lib/jasmine-pit.js

```diff
@@ -5,82 +5,55 @@
  * LICENSE file in the root directory of this source tree. An additional grant
  * of patent rights can be found in the PATENTS file in the same directory.
  */
+
+ /**
+ * This module adds ability to test async promise code with jasmine by
+ * returning a promise from `it\fit` block.
+ */
+
 'use strict';
 
-function assertPromise(promise) {
-  if (!promise || typeof promise.then !== 'function') {
-    throw new Error(`pit(...): tests must return a promise`);
-  }
+function isPromise(obj) {
+  return obj && typeof obj.then === 'function';
 }
 
-function wrapForJasmine1(env, promiseFn) {
-  return () => {
-    let error = null;
-    let isFinished = false;
-    const currentSpec = env.currentSpec;
-    currentSpec.runs(() => {
-      try {
-        const promise = promiseFn.call(currentSpec);
-        assertPromise(promise);
-        promise
-          .then(() => isFinished = true)
-          .catch(err => {
-            error = err;
-            isFinished = true;
-          });
-      } catch (e) {
-        error = e;
-        isFinished = true;
-      }
-    });
-    currentSpec.waitsFor(() => isFinished);
-    currentSpec.runs(() => {
-      if (error) {
-        throw error;
-      }
-    });
-  };
-}
+// return a wrapping function: `env.fit = makePromiseIt(env, 'fit')`
+function makePromiseIt(env, fnName) {
+  const originalIt = env[fnName];
 
-function wrapForJasmine2(promiseFn) {
-  return done => {
-    const promise = promiseFn();
-    const fail = done.fail ? done.fail : err => { throw err; };
+  if (typeof originalIt !== 'function') {
```

Comment:
I don't think we really need this error message tbh, this stuff is only local to this file.